pkgs/development/libraries/catch2/3.nix

```diff
@@ -31,13 +33,13 @@ stdenv.mkDerivation rec {
     "-DCMAKE_CTEST_ARGUMENTS=-E;ApprovalTests"
   ];
 
-  env = lib.optionalAttrs stdenv.hostPlatform.isx86_32 {
+  env.NIX_CFLAGS_COMPILE = toString([]
     # Tests fail on x86_32 if compiled with x87 floats: https://github.com/catchorg/Catch2/issues/2796
-    NIX_CFLAGS_COMPILE = "-msse2 -mfpmath=sse";
-  } // lib.optionalAttrs (stdenv.hostPlatform.isRiscV || stdenv.hostPlatform.isAarch32) {
+    ++ lib.optional stdenv.hostPlatform.isx86_32 "-msse2 -mfpmath=sse"
     # Build failure caused by -Werror: https://github.com/catchorg/Catch2/issues/2808
-    NIX_CFLAGS_COMPILE = "-Wno-error=cast-align";
-  };
+    ++ lib.optional (stdenv.hostPlatform.isRiscV || stdenv.hostPlatform.isAarch32) "-Wno-error=cast-align"
+    ++ lib.optional stdenv.cc.isClang "-Wno-error=c++20-extensions"
```

Comment:
yeah, maybe but I didn't want to change the behavior from upstream. I'm not going to test it to see if using c++20 is going to break their coding assumptions.

Reply:
and the base code builds fine without it its just that we compile with `CATCH_DEVELOPMENT_BUILD=ON` for some reason which adds the code that triggers the error, IIRC, so disabling the warning seemed minimally invasive.

Reply:
They support C++20, e.g.

* https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/docs/release-notes.md?plain=1#L326-L328
* https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/src/catch2/internal/catch_meta.hpp#L31-L33
* https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/tests/SelfTest/UsageTests/Message.tests.cpp#L244-L245

But I guess we turn on the development build because [it enables tests](https://github.com/catchorg/Catch2/blob/7bbd4b90750ea20a08377c600da924d5d6d7f8ef/docs/cmake-integration.md). We should just toggle `CATCH_ENABLE_WERROR` back off, since that’s the default behaviour otherwise and it’s harmful rather than useful for us.